src/datastore/datastore.rs

```diff
@@ -253,14 +244,31 @@ impl DatastoreWorker {
 }
 
 impl DatastoreInstance {
-
     fn get_stored_buckets(&mut self, conn: &Connection) -> Result <(), DatastoreError> {
-        let mut stmt = match conn.prepare("SELECT id, name, type, client, hostname, created, data FROM buckets") {
+        let mut stmt = match conn.prepare("
+            SELECT  buckets.id, buckets.name, buckets.type, buckets.client,
+                    buckets.hostname, buckets.created, buckets.data,
+                    min(events.starttime), max(events.endtime)
+            FROM buckets
+            INNER JOIN events ON buckets.id = events.bucketrow
+            GROUP BY buckets.id
+            ;") {
```

Comment:
Is this query still about as fast? (I'm kinda bad at predicting performance of SQL queries)

Reply:
Well, I'm no expert either, but these are my two cents: Firstly inner joins and group by are made to support very large data sets in real SQL servers and this should be negligible there, but if sqlite works differently I don't know. Secondly we only have a few buckets. Thirdly this query only runs once when the data store is initialized, after that the buckets are cached. Lastly I don't have any tests for this yet so I might have to modify this query if it doesn't work as expected. This PR need more tests.

Reply:
Another point is that starttime and endtime is indexed, so to get those values should be pretty much instant.